### PR TITLE
FullyQualifiedStrictTypesFixer - fix same classname is imported from …

### LINKED ISSUE #4726: fully_qualified_strict_types messes up when same classname is imported from global and non-global namespace
Hey there. `fully_qualified_strict_types` does strange things when a class with same name is imported from both the global namespace and a non-global one.

#### The PHP version you are using (`$ php -v`):
=> 7.1.30

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> 2.16.1

#### The command you use to run PHP CS Fixer:
=> ./vendor/bin/php-cs-fixer fix -v

#### The configuration file you are using, if any:
```php
=> fully_qualified_strict_types => true
```

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
namespace App\Models {
    class TestModel {}
}

namespace {
    class TestModel {}
}

namespace App {
    use App\Models\TestModel;
    use TestModel as TestModelAlias;

    class TestTest
    {
        public function test(): TestModel
        {
            TestModelAlias::class;
        }
    }
}
```

 * with unexpected changes applied when running PHP CS Fixer:
```php
- public function test(): TestModel
+ public function test(): TestModelAlias
```

 * with the changes you expected instead:
=> None

### PR DESCRIPTION
**Pull request by SpacePossum**
…global and non-global namespace

closes #4726 

there are multiple issues with this fixer, please have a look at the new logic and tests

### CODE REVIEW

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php [hunk 1]
```diff
@@ -391,15 +508,15 @@ function doSomething(\Foo\Bar\SomeClass $foo, \Foo\Bar\Buz $buz, \Foo\Bar\Zoof\B
 class SomeClass
 {
     public function doSomething(
-        Ping\Something $something,
-        Pung\Pang $pungpang,
-        Pung $pongpung,
-        Pang\Pung $pangpung,
-        Pyng\Pung\Pong $pongpyngpangpang,
-        Bar\Baz\Buz $bazbuz
+        \Ping\Something $something,
```
Comment (SpacePossum) [inline tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:511]: I think the current fixer is wrong here, removing the leader `\` within the current namespace changes the type that is supposedly used here

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php [hunk 2]
```diff
@@ -391,15 +508,15 @@ function doSomething(\Foo\Bar\SomeClass $foo, \Foo\Bar\Buz $buz, \Foo\Bar\Zoof\B
...
+        Pung\Pang $other,
+        Pung $other1,
+        Pang\Pung $other2,
+        Pyng\Pung\Pong $other3,
+        \Foo\Bar\Baz\Buz $other4
```
Comment (SpacePossum) [inline tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:516]: same here

**Comment by SpacePossum** [comment]
note to self; make coverage report of the fixer to see if all paths are covered

**Review by kubawerlos — CHANGES_REQUESTED** [review]
Let's add a test case for:
```php
<?php
namespace Foo\Bar;
function test(\Foo\Bar $x) {}
```

**Comment by SpacePossum** [comment]
nice edge-case, thanks :+1:  , patched

**Review by kubawerlos — APPROVED** [review]

**Pull request closed without merging**